### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by gpeacock**
## Changes in This Pull Request
Some documentation cleanup work

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [x] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

**Review by scouten-adobe — CHANGES_REQUESTED** [review]
Looks good, but I'd like to make a few editing changes. Mind holding this until this evening when I can focus on this PR?

**Comment by gpeacock** [comment]
This makes the Ingredient, Thumbnail and DataHash assertions crate private. As a result, some of the APIs that are not used internally are now unused code. We could strip out that code, but I expect at some point we may make these public as a lower level API. Hence the allow-unused additions.

**Review by mauricefisher64 — DISMISSED** [review]
👍

File: sdk/src/assertion.rs
Comment (scouten-adobe) [inline sdk/src/assertion.rs:205]: As we discussed on Slack, not sure what to do with this since `AssertionData` is not exposed from the crate.

**Review by scouten-adobe — CHANGES_REQUESTED** [review]
I've made the edits I want to make here; would still like to see the dead-code attributes removed before merging.

File: sdk/src/assertion.rs
Comment (gpeacock) [reply sdk/src/assertion.rs:205]: I changed the references to AssertionData to not be in public comments, since it is not exposed

**Comment by gpeacock** [comment]
I discussed this with Maurice. Since the plan is to eventually make these Assertions public again at some point, we would rather not delete all the code just to make Clippy happy for now. I make the allow(dead_code) statements at the module level with comments as to why we are doing it.

File: sdk/src/assertion.rs
Comment (gpeacock) [reply sdk/src/assertion.rs:205]: I just removed the public comments about AssertionData so they don't show up in the docs - also corrected this so that the first line is only one line and not a big paragraph.

**Review by scouten-adobe — APPROVED** [review]

**Pull request merged**
